File: app/src/main/java/com/chiranths/jobportal1/Adapters/LivingPlaceAdaptor.java
package com.chiranths.jobportal1.Adapters;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.chiranths.jobportal1.Activities.HotDealsactivity.HotDealsDetailsActivity;
import com.chiranths.jobportal1.Model.LivingPlaceModel;
import com.chiranths.jobportal1.R;

import java.util.List;

public class LivingPlaceAdaptor extends RecyclerView.Adapter<LivingPlaceAdaptor.ViewHolder> {


    private List<LivingPlaceModel> productInfos;
    private Context context;


    public LivingPlaceAdaptor(List<LivingPlaceModel> productInfos, Context context) {
        this.productInfos = productInfos;
        this.context = context;
    }

    @NonNull
    @Override
    public LivingPlaceAdaptor.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        context = parent.getContext();
        View listItem= layoutInflater.inflate(R.layout.recyclar_livingplace_layout, parent, false);
        LivingPlaceAdaptor.ViewHolder viewHolder = new LivingPlaceAdaptor.ViewHolder(listItem);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        LivingPlaceModel productInfo = productInfos.get(position);

        Glide.with(context)
                .load(productInfo.getImage())
                .into(holder.iv_living_image);

        holder.tv_living_title.setText(productInfo.getTitle());
        holder.tv_living_rent_amount.setText("Rent: "+productInfo.getRentamount());
        holder.tv_living_sqft.setText(productInfo.getSqft());
        holder.tv_number_bhk.setText("Info: "+productInfo.getNuBHK());
        holder.tv_loaction_living.setText(productInfo.getLocation());

        holder.cv_card_living.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent =new Intent(context, HotDealsDetailsActivity.class);
                intent.putExtra("pid",productInfo.getPid());
                context.startActivity(intent);
            }
        });

    }

    @Override
    public int getItemCount() {
        return productInfos.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        ImageView iv_living_image,iv_call_bottom,iv_whatsapp_bottom;
        CardView cv_card_living;
        TextView tv_living_title,tv_living_rent_amount,tv_living_sqft,tv_loaction_living,tv_number_bhk;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            iv_living_image = itemView.findViewById(R.id.iv_living_image);
            cv_card_living = itemView.findViewById(R.id.cv_card_living);
            tv_living_title = itemView.findViewById(R.id.tv_living_title);
            tv_living_rent_amount = itemView.findViewById(R.id.tv_living_rent_amount);
            tv_living_sqft = itemView.findViewById(R.id.tv_living_sqft);
            tv_loaction_living = itemView.findViewById(R.id.tv_loaction_living);
            tv_number_bhk = itemView.findViewById(R.id.tv_number_bhk);
        }
    }
}

File: app/src/main/java/com/sbd/gbd/Adapters/LivingPlaceAdaptor.kt
package com.sbd.gbd.Adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.google.firebase.database.FirebaseDatabase
import com.sbd.gbd.Activities.BasicActivitys.RentHomeDetails
import com.sbd.gbd.Model.LivingPlaceModel
import com.sbd.gbd.R
import com.sbd.gbd.Utilitys.AppConstants
import com.sbd.gbd.Utilitys.Utilitys

class LivingPlaceAdaptor(
    private val productInfos: List<LivingPlaceModel>,
    private var context: Context
) : RecyclerView.Adapter<LivingPlaceAdaptor.ViewHolder>() {
    var utilitys = Utilitys()

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        context = parent.context
        val listItem =
            layoutInflater.inflate(R.layout.recyclar_livingplace_layout, parent, false)
        return ViewHolder(listItem)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val productInfo = productInfos[position]
        Glide.with(context)
            .load(productInfo.image)
            .into(holder.iv_living_image)
        holder.tv_living_title.text = productInfo.title
        holder.tv_living_rent_amount.text =  productInfo.rentamount
        holder.tv_living_sqft.text = productInfo.sqft
        holder.tv_number_bhk.text = productInfo.nuBHK + "BHK"
        holder.tv_loaction_living.text = productInfo.location
        holder.cv_card_living.setOnClickListener {
            val intent = Intent(context, RentHomeDetails::class.java)
            intent.putExtra(AppConstants.pid, productInfo.pid)
            context.startActivity(intent)
        }
        if (productInfo.status.equals("1")) {
            holder.ll_approve.visibility = View.VISIBLE
            holder.ll_call?.visibility = View.GONE
        } else if (productInfo.status.equals("3")) {
            holder.ll_approve.visibility = View.VISIBLE
            holder.ll_call?.visibility = View.GONE
        } else {
            holder.ll_approve.visibility = View.GONE
            holder.ll_call?.visibility = View.VISIBLE
        }

        holder.ll_approve.setOnClickListener {
            productInfo.pid?.let { it1 ->
                FirebaseDatabase.getInstance().reference.child("livingplaceforyou").child(it1)
                    .child(AppConstants.Status).setValue("2")
            }
        }
        holder.iv_const_call_bottom.setOnClickListener { view: View? ->
            utilitys.navigateCall(
                context,
                productInfo.contactNumber,
                productInfo.postedBy
            )
        }
        holder.iv_const_whatsapp_bottom.setOnClickListener { view: View? ->
            utilitys.navigateWhatsapp(
                context,
                productInfo.contactNumber,
                productInfo.postedBy
            )
        }
    }

    override fun getItemCount(): Int {
        return productInfos.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var iv_living_image: ImageView
        var iv_call_bottom: ImageView? = null
        var iv_whatsapp_bottom: ImageView? = null
        var cv_card_living: CardView
        var ll_approve: LinearLayout
        var ll_call: LinearLayout
        var tv_living_title: TextView
        var tv_living_rent_amount: TextView
        var tv_living_sqft: TextView
        var tv_loaction_living: TextView
        var tv_number_bhk: TextView
        var iv_const_whatsapp_bottom: ImageView
        var iv_const_call_bottom: ImageView

        init {
            iv_living_image = itemView.findViewById(R.id.iv_living_image)
            cv_card_living = itemView.findViewById(R.id.cv_card_living)
            tv_living_title = itemView.findViewById(R.id.tv_living_title)
            tv_living_rent_amount = itemView.findViewById(R.id.tv_living_rent_amount)
            tv_living_sqft = itemView.findViewById(R.id.tv_living_sqft)
            tv_loaction_living = itemView.findViewById(R.id.tv_loaction_living)
            tv_number_bhk = itemView.findViewById(R.id.tv_number_bhk)
            ll_approve = itemView.findViewById(R.id.ll_approve)
            ll_call = itemView.findViewById(R.id.ll_call)
            iv_const_whatsapp_bottom = itemView.findViewById(R.id.iv_whatsapp_bottom)
            iv_const_call_bottom = itemView.findViewById(R.id.iv_call_bottom)
        }
    }
}
